src/comps/temp-withdraw.tsx

```diff
@@ -29,65 +24,25 @@ import {
   callContractXverse,
 } from "@/util/wallet-utils";
 import { getStacksNetwork } from "@/util/get-stacks-network";
+import { validate, Network, getAddressInfo } from "bitcoin-address-validation";
+import { decodeBitcoinAddress } from "@/util/decode-bitcoin-address";
 
 const decodeBitcoinAddressToClarityRecipient = (
   address: string,
   network: DefaultNetworkConfigurations,
 ) => {
-  const lower = address.toLowerCase();
   const isMainnet = network === "mainnet";
-
-  // check if passed in address is bech32
-  let decodedBech32;
-  let isBech32 = false;
-  if (lower.startsWith("bc1") || lower.startsWith("bcrt1")) {
-    try {
-      decodedBech32 = bitcoin.address.fromBech32(address);
-      isBech32 = true;
-    } catch (err: any) {
-      throw new Error(`Invalid bech32 address: ${err.message}`);
-    }
-  }
-
-  if (isBech32 && decodedBech32) {
-    const {
-      prefix,
-      version: witnessVersion,
-      data: witnessProgram,
-    } = decodedBech32;
-
-    const hexHash = "0x" + bytesToHex(witnessProgram);
-    if (isMainnet) {
-      if (prefix !== "bc") {
-        throw new Error(`Expected mainnet bech32 (bc1...), got ${prefix}`);
-      }
-    } else if (!isMainnet) {
-      // regtest typically uses prefix 'bcrt'
-      if (prefix !== "bcrt") {
-        throw new Error(`Expected regtest bech32 (bcrt1...), got ${prefix}`);
-      }
-    }
-
-    if (witnessVersion === 0) {
-      if (witnessProgram.length === 20) {
-        // P2WPKH
-        return { version: "0x04", hashbytes: hexHash };
-      } else if (witnessProgram.length === 32) {
-        // P2WSH
-        return { version: "0x05", hashbytes: hexHash };
-      } else {
-        throw new Error(
-          `Unsupported witness v0 length=${witnessProgram.length}`,
-        );
-      }
-    } else if (witnessVersion === 1 && witnessProgram.length === 32) {
-      // P2TR
-      return { version: "0x06", hashbytes: hexHash };
-    } else {
-      throw new Error(
-        `Unrecognized witnessVersion=${witnessVersion}, length=${witnessProgram.length}`,
-      );
-    }
+  const addressInfo = getAddressInfo(address);
+  let addressNetwork = isMainnet ? Network.mainnet : Network.regtest;
+  // bech32 is different between all three networks
+  // base58 is same between regtest and testnet
+  const isRegtestBase58 = !addressInfo.bech32 && !isMainnet;
+  addressNetwork = isRegtestBase58 ? Network.testnet : addressNetwork;
```

Comment:
Why do we need these 4 lines? We only care about mainnet and regtest, so the `addressNetwork = isMainnet ? Network.mainnet : Network.regtest;` above should cover all we need.
Then if you supply a testnet address it would fail validation, which is fine as we don't work on testnet.

Reply:
the validation lib defaults to testnet when using base58 addresses the same address is valid for regtest, as explained in the comments, so it would break if the passed address is regtest base58 and the network is regtest

Reply:
Ah got it. Can you expand a bit the comment to make it clear what's happening? Basically as you did in this convo